### ISSUE TITLE
When SCHEDULE_FOR_RP_ONLY is YES, schedule length affects cost function lower bound

### ISSUE THREAD

**Issue by Quincunx271** [CSUS-LLVM/OptSched#43]
When SCHEDULE_FOR_RP_ONLY is YES, the schedule length still impacts the cost function at a minimum in the lower bound calculation. In particular, `CmputCostLwrBound()` scales the schedule length by `schedCostFactor_` in bb_spill.cpp, and `schedCostFactor_` is only ever set to `COST_WGHT_BASE` which is `10`.

My SPILL_COST_FUNCTION is PERP. I did neglect to set LATENCY_PRECISION away from LLVM, but I was under the impression that SCHEDULE_FOR_RP_ONLY would imply a LATENCY_PRECISION of UNIT.

**Comment by kerbowa** [comment]
The issue is with the meaning of SCHEDULE_FOR_RP_ONLY. We added an enumerator pruning technique with this name. Right now that setting is only referring to that pruning technique. The idea is that this technique can only be used when RP is the only thing that matters.

The problem is that there are many things we can actually do when we are scheduling for RP only. I've implemented things in private branches like not doing relaxed scheduling ect in the RP pass. What you have suggested also fits into that category.

The RP-only pruning which I believe @vangthao95 was working with may be buggy. I'm unsure whether it should always be enabled in the RP pass. The current default behavior is to not enable this.

I propose that we create another option called RP_ONLY_PRUNING that has the same current effects of SCHEDULE_FOR_RP_ONLY. We then can update the latter flag to contain all the safe improvements when RP is the only thing we are about.
